types/metro-resolver/FailedToResolvePathError.d.ts

```diff
@@ -0,0 +1,6 @@
+import { FileAndDirCandidates } from './types';
```

Comment:
Can we use `import type` here or did that fail CI?
```suggestion
import type { FileAndDirCandidates } from './types';
```

Reply:
I have a question, all those stuff are already type-only, why do we still need to use the `import type` keyword instead of `import`?

Reply:
The CI checking is already passed: https://github.com/DefinitelyTyped/DefinitelyTyped/actions/runs/3465984684/jobs/5789339978

Reply:
It's mostly to make the parser's job easier. For example, Babel can simply drop this statement entirely since it doesn't care about types.

Reply:
Babel was an example. It also helps TS and affects the output. For example, it prevents people from using a type as a class. There's an example of that here: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export

Reply:
> For example, it prevents people from using a type as a class.

From my understanding, it prevents people from using a type as a class, is because the type is imported by `import type`, so the `tsc` will only import the "type context value" into the module. But in our case, our file is `.d.ts`, so I think the `tsc` can only import the "type context value" into the module, it's almost identical to `import type` a class from a `.ts` file

Does my understanding correct? Or maybe I missed something?

Reply:
No, I don't think that's right. By that logic, it wouldn't be possible to do `new FailedToResolveNameError()` because you're importing it from a `.d.ts` file. Why would importing from a `.d.ts` file change the type? That doesn't make any sense.